insights/parsers/satellite_postgresql_query.py

```diff
@@ -0,0 +1,196 @@
+            search criteria.
+
+        Examples::
+
+            >>> query.search(name__startswith='abc') == [
+            ... {'name': 'abc', 'url': '', 'value': 'test'},
+            ... {'name': 'abcdef', 'url': '', 'value': 'test2'}
+            ... ]
+            True
+            >>> query.search(name__startswith='abc', value='test') == [
+            ... {'name': 'abc', 'url': '', 'value': 'test'}
+            ... ]
+            True
+        """
+
+        return keyword_search(self, **kwargs)
+
+
+@parser(Specs.satellite_settings)
+class SatelliteAdminSettings(SatellitePostgreSQLQuery):
+    """
+    Parse the output of the command ``psql -d foreman -c '"select name, value, "default" from settings where name in ('destroy_vm_on_host_delete', 'unregister_delete_host') --csv"``.
+
+    Sample output::
+
+        name,value,default
+        unregister_delete_host,"--- true
+        ...","--- false
+        ..."
+        destroy_vm_on_host_delete,,"--- true
+        ..."
+
+    Examples::
+
+        >>> type(table)
+        <class 'insights.parsers.satellite_postgresql_query.SatelliteAdminSettings'>
+        >>> table.has_setting('destroy_vm_on_host_delete')
+        True
+        >>> table.get_setting('unregister_delete_host')
+        True
+        >>> table.get_setting('destroy_vm_on_host_delete')
+        True
+    """
+
+    def _parse_yaml(self, value):
+        if value:
+            try:
+                return yaml.safe_load(value)
+            except Exception:
+                raise ParseException("Bad format value: %s" % value)
+        return value
+
+    def parse_content(self, content):
+        """
+        The "default" and "value" columns must be selected, or else the
+        settings value can't be determined.
+        The "default" and "value" column are in yaml format, it is transfer to
+        python object.
+
+        Raises::
+
+            SkipException: when value or default column isn't found in the
+                            table.
+            ParseException: when the value or default in bad yaml format.
+        """
+        super(SatelliteAdminSettings, self).parse_content(content)
+        if not all(item in self.get_columns() for item in ['default', 'value']):
+            raise SkipException('No default, value columns in the table.')
+        for row in self:
+            row['default'] = self._parse_yaml(row['default'])
+            row['value'] = self._parse_yaml(row['value'])
+
+    def has_setting(self, setting_name):
+        """Return True if the setting_name exists in the table otherwise False"""
+        rows = self.search(name=setting_name)
+        return True if rows else False
+
+    def get_setting(self, setting_name):
+        """
+        Get the actual value of setting_name.
+        If the value column isn't empty, the value of the setting_name is the
+        value column, or else it's the default column.
+
+        Args::
```

Comment:
The two colons `::` is not necessary for these proper sections.

see: https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html